src/toxcore/dht/server.rs

```diff
@@ -983,4 +1005,34 @@ mod tests {
         let (_, lan_discovery) = LanDiscovery::from_bytes(&buf[..size]).unwrap();
         assert_eq!(alice.pk, lan_discovery.pk);
     }
+    // remove_timedout_clients(), case of client removed
+    #[test]
+    fn server_remove_timedout_clients_removed_test() {
+        let (alice, _precomp, bob_pk, _bob_sk, _rx, _addr) = create_node();
+        let node = PackedNode::new(false, SocketAddr::V4("127.0.0.1:12345".parse().unwrap()), &bob_pk);
+
+        alice.try_add_to_kbucket(&node);
+
+        alice.send_pings().wait().unwrap();
+        let dur = time::Duration::from_secs(2);
+        thread::sleep(dur);
```

Comment:
Maybe we shouldn't freeze test at all? Just use alice.remove_timedout_clients(0).

Reply:
Namsoo, try it. But it is optional since 2 secs != 122 =)